src/Payum/Action/HyperPayAction.php

```diff
@@ -96,7 +146,11 @@ public function execute($request, $client, $api)
 
         try {
             $payment->setDetails(['status' => PaymentInterface::STATE_PROCESSING]);
-            $paymentToken = $this->getcheckoutId($order, number_format($payment->getOrder()->getTotal() / 100, 2));
+            if ($payment->getMethod()->getCode() === 'tamara')  {
```

Comment:
@ahmed-omani Please we don't need this kind of implementation if and else, use an appropriate design pattern